pkg/impersonation/impersonation.go

```diff
@@ -182,10 +183,44 @@ func (i *Impersonator) createRole() (*rbacv1.ClusterRole, error) {
 			// in case cache isn't synced yet, use raw client
 			return i.clusterContext.RBAC.ClusterRoles("").Get(name, metav1.GetOptions{})
 		}
+		return r, nil
+	}
+	if !reflect.DeepEqual(rules, r.Rules) {
+		return i.updateRoleRules(r, rules[1], rules[2], rules[3])
```

Comment:
is there anything that guarentees that rules[2] and result[3] won't result in a panic?

Reply:
nope. fixed.